cartographer_ros/cartographer_ros/node_options.cc

```diff
@@ -34,6 +36,10 @@ NodeOptions CreateNodeOptions(
       lua_parameter_dictionary->GetDouble("submap_publish_period_sec");
   options.pose_publish_period_sec =
       lua_parameter_dictionary->GetDouble("pose_publish_period_sec");
+  options.trajectory_publish_period_sec =
+      lua_parameter_dictionary->HasKey("trajectory_publish_period_sec")
```

Comment:
Yes, please make it mandatory. While it is some work to fix all configurations, we are pre-1.0 and breakage is expected. If we are too much concerned about breaking existing users, we would quickly accumulate even more technical debt then we already have.